apps/judicial-system/web/src/components/PdfButton/PdfButton.tsx

```diff
@@ -32,12 +36,26 @@ const PdfButton: React.FC<Props> = ({
   children,
   // Overwrites the default onClick handler
   handleClick,
+  policeCaseNumber,
 }) => {
-  const handlePdfClick = () => {
-    window.open(
-      `${api.apiUrl}/api/case/${caseId}/${pdfType}?useSigned=${useSigned}`,
-      '_blank',
-    )
+  const [isPdfLoading, setIsPdfLoading] = useState<boolean>(false)
+  const { formatMessage } = useIntl()
+
+  const handlePdfClick = async () => {
+    try {
+      const newPdfType =
+        pdfType === 'caseFiles' ? `${pdfType}/${policeCaseNumber}` : pdfType
+      const url = `${api.apiUrl}/api/case/${caseId}/${newPdfType}?useSigned=${useSigned}`
+
+      setIsPdfLoading(true)
+
+      await fetch(url)
+
+      setIsPdfLoading(false)
+      window.open(url, '_blank')
```

Comment:
This will result in us generating the pdf 2x. Unless the backend handles this for us or the browser by caching the pdf.